### ISSUE TITLE
No subtitles error is back after fix

### ISSUE THREAD

**Issue by ToborWinner** [justchokingaround/lobster#233]
**Describe the bug**
Whenever I try to play something using the most recent version of lobster, it shows "No subtitles found" and exits after selecting the movie/episode. Last time it was because the API server was down. During the time when the API was still down I self-hosted lobster-netlify-api and it worked just fine. Now, even if I switch to the self-hosted version, it fails. The self-hosted api shows an error saying that in rabbit.ts at line 720 it's trying to access element 0 of null.

Run the lobster command, select any movie and try to play it.

**Expected behavior**
The API should work and the movie should play.

**Desktop**
 - OS: NixOS

**Comment by DiegoOlaya** [comment]
I am experiencing the same issue on MacOS Sonoma 14.7. The program will exit with "No subtitles found" upon selecting any piece of media.

Steps to reproduce:
Run `lobster` with any episode.

Expected:
The media player should open and the episode should play.

**Comment by EasperOne** [comment]
Same here on Ubuntu 24.04.1. The error persisted no matter I self-hosted or not.

**Comment by Mra1k3r0** [comment]
up

**Comment by pitsi** [comment]
Same issue here. 

If someone knows how to use this to decrypt what is needed, please post. Also post what the url for json_data must change to. My braincells have only succeded to "npm install" in it and nothing more to make it run.
https://github.com/justchokingaround/rabbit_wasm_api

**Comment by pitsi** [comment]
I made some progress! You clone the forementioned repo, cd into it, run `npm install` to let it install anything it wants and then run `npm run start`. Proof
```
# npm run start

> rabbit_wasm@1.0.0 start
> ts-node index.ts

Server is running on port 3000
```
Now someone must find the proper url to use under the json_data variable in the script, because this does not work
`http://server-ip-or-name:3000/.netlify/functions/decrypt?id=${source_id}`

**Comment by pitsi** [comment]
@FurqanHun
Sorry for adding you in the conversation, but I think you may have an idea on how to run it.

**Comment by FurqanHun** [comment]
> @FurqanHun Sorry for adding you in the conversation, but I think you may have an idea on how to run it.

Yeah, I know about it and actually fixed it in my own fork.

In `decrypt.ts` at line 31, it says:

```ts
async function decrypt(source: string) {
  // Await the async function call
  return await main(source);
}
```

You’ll need to add an argument and change it to:

```ts
async function decrypt(source: string, provider: string) {
  // Await the async function call
  return await main(provider, source);
}
```

Locally, you’ll just get a warning, but it still works since the `provider` is hard-coded in `rabbit.ts`. Netlify, though, won’t let you get away with that. You can clone [my fork](https://github.com/FurqanHun/lobster-netlify-api) and deploy it directly; that should work.

# MUST READ THIS PART!!!
But even after deploying and updating the config file, it might not work because the **script’s broken again**. The author mentioned (on discord/matrix) that rabbit stream changed their encryption thingy, which messed things up. So, you might need to wait for a fix or help out with it. Also, refrain from creating issues about it as it's already a known issue.

**Comment by pitsi** [comment]
I just tried your fork and it does not work. Proof from lobster -x
```
+ curl -s http://alpine.local:3000/.netlify/functions/decrypt?id=RSUL5z8uzcgO
+ json_data=
```
I saw npm mention 3999 as the port while it was launching netlify, so I tried it too, but nothing changed. Thanks a lot though for patching it!

**Comment by FurqanHun** [comment]
> I just tried your fork and it does not work.

Yep, i did mentioned it still won't work as lobster's broken rn, and there's no ETA for when it will be fixed. If you know how to reverse engineerband re code the rabbit stream thing you can try contributing tho.

**Comment by pitsi** [comment]
@FurqanHun 
On a system that I do not WANT to install ts-node system-wide, how can I make npm run a .ts script?
Asking for a friend and thank you in advance :D

**Comment by FurqanHun** [comment]
> @FurqanHun On a system that I do not WANT to install ts-node system-wide, how can I make npm run a .ts script? Asking for a friend and thank you in advance :D

To install locally you just omit the `-g` or `--global` flag with the install command:

```bash
npm install ts-node typescript
```
However, it’s recommended to use the `--save-dev` flag to add the packages as development dependencies:

```bash
npm install --save-dev ts-node typescript
```

After that, you'd need to add your `.ts` script in `package.json`:
```json
{
  "scripts": {
    "start": "ts-node your_script.ts"
  }
}
```

and then you can just run it with:
```bash
npm run start
```

**Comment by FurqanHun** [comment]
@pitsi 

If you haven't set up the `tsconfig.json`, that might be why it's not working. `ts-node` auto compiles your ts code before running it, and having the right config is key since there are different ways to compile... especially with various syntax.

I did test it myself btw:

`script.ts`
```ts
console.log("Testing typescript with node-ts...");
```

Added the following in `package.json`
```json
"scripts": {
    "start": "ts-node script.ts",
  },
```
Created `tsconfig.json`:
```json
{
  "compilerOptions": {
    "target": "ES6",
    "module": "commonjs",
    "strict": true
  },
  "include": ["*.ts"],
  "exclude": ["node_modules"]
}

```
and here's the output:
![image](https://github.com/user-attachments/assets/17df7faf-d623-4924-8c69-50a75c99b66d)

_`npm list --depth=0` shows the installed packages of the project and `-g` does the same but for global_

Feel free to reach out to me at business.furqanhun@gmail.com or on discord at `furqanhun` as going back and forward here just notifies everyone and this isn't related to the project either...

**Comment by Yuzuru-Yamazaki** [comment]
still not fixed??
